pkg/netpol/connlist/connlist.go

```diff
@@ -383,27 +422,64 @@ func (ca *ConnlistAnalyzer) getConnectionsList(pe *eval.PolicyEngine) ([]Peer2Pe
 			}
 			allowedConnections, err := pe.AllAllowedConnectionsBetweenWorkloadPeers(srcPeer, dstPeer)
 			if err != nil {
-				ca.errors = append(ca.errors, newResourceEvaluationError(err))
 				return nil, err
 			}
 			// skip empty connections
 			if allowedConnections.IsEmpty() {
 				continue
 			}
-			connectionObj := &connection{
+			p2pConnection := &connection{
 				src:               srcPeer,
 				dst:               dstPeer,
 				allConnections:    allowedConnections.AllConnections(),
 				protocolsAndPorts: allowedConnections.ProtocolsAndPortsMap(),
 			}
-			res = append(res, connectionObj)
+			res = append(res, p2pConnection)
+		}
+	}
+	return res, nil
+}
+
+// getIngressAllowedConnections returns connections list from IngressAnalyzer intersected with PolicyEngine's connections
+func (ca *ConnlistAnalyzer) getIngressAllowedConnections(ia *ingressanalyzer.IngressAnalyzer,
+	pe *eval.PolicyEngine) ([]Peer2PeerConnection, error) {
+	res := make([]Peer2PeerConnection, 0)
+	ingressConns, err := ia.AllowedIngressConnections()
+	if err != nil {
+		return nil, err
+	}
+	// adding the ingress controller pod to the policy engine,
+	ingressControllerPod, err := pe.AddPodByNameAndNamespace(ingressanalyzer.IngressPodName, ingressanalyzer.IngressPodNamespace)
+	if err != nil {
+		return nil, err
+	}
+	for peerStr, peerAndConn := range ingressConns {
+		// compute allowed connections based on pe.policies to the peer, then intersect the conns with
+		// ingress connections to the peer -> the intersection will be appended to the result
+		peConn, err := pe.AllAllowedConnectionsBetweenWorkloadPeers(ingressControllerPod, peerAndConn.Peer)
+		if err != nil {
+			return nil, err
+		}
+		peerAndConn.ConnSet.Intersection(peConn.(*common.ConnectionSet))
+		if peerAndConn.ConnSet.IsEmpty() {
+			ca.logger.Warnf("Analysis of Ingress/Route resources inferred there is a configured connectivity from ingress-controller to workload " +
+				peerStr + ", but network policies may be blocking access to this workload" +
+				"(unless specifically permitted to the installed ingress controller pod)")
+			continue
+		}
```

Comment:
change warning as follows:
`Ingress/Route (name) specified workload (name) as a backend, but network policies are  blocking ingress connections from an arbitrary in-cluster source to this workload. Connectivity map will not include a possibly allowed connection between the ingress controller and this workload.`

replace `(name)` with the relevant resource name

Reply:
updated, however
we don't have an indication from here of the ingress/route name/s which selects the workload (+ more than one ingress/route objects may select same service (mapped to same peer)).

Reply:
can you collect this information to be displayed in this message?

Reply:
yes its possible, updating

Reply:
let's be consistent with resource naming in the same format